rust/src/lib/ifaces/hsr.rs

```diff
@@ -65,6 +75,89 @@ impl HsrInterface {
     }
 }
 
+impl MergedInterfaces {
+    pub(crate) fn get_hsr_mac(&self, iface: &HsrInterface) -> Option<String> {
+        if iface.base.mac_address.is_some() {
+            return iface.base.mac_address.clone();
+        }
+
+        let hsr_conf = iface.hsr.as_ref()?;
+
+        for use_permanent in [false, true] {
+            for port in [&hsr_conf.port1, &hsr_conf.port2] {
+                let mac = self
+                    .get_iface(port, InterfaceType::Unknown)
+                    .and_then(|iface| {
+                        let base = iface.merged.base_iface();
+                        if use_permanent {
+                            base.permanent_mac_address.clone()
+                        } else {
+                            base.mac_address.clone()
+                        }
+                    });
+                if mac.is_some() {
+                    return mac;
+                }
+            }
+        }
```

Comment:
I don't really understand why this work... you said that mac-address was always defined in the merged interface, so it would always pick the address from port1, never from port2. Clearly I'm missing something because you have a test for that...